tests/components/Geolocation.test.tsx

```diff
@@ -0,0 +1,230 @@
+
+jest.mock('@yext/search-headless-react');
+
+const mockedState: Partial<State> = {
+  filters: {
+    static: []
+  },
+  vertical: {
+    verticalKey: 'jobs',
+  },
+  searchStatus: {
+    isLoading: false
+  },
+  meta: {
+    searchType: 'vertical'
+  }
+};
+
+const mockedStateWithFilters: Partial<State> = {
+  ...mockedState,
+  filters: {
+    static: [
+      {
+        displayName: 'Some Location',
+        selected: true,
+        filter: {
+          kind: 'fieldValue',
+          fieldId: 'builtin.location',
+          matcher: Matcher.Near,
+          value: {
+            lat: 1,
+            lng: 1,
+            radius: 10
+          },
+        }
+      },
+      {
+        displayName: 'Current Location',
+        selected: true,
+        filter: {
+          kind: 'fieldValue',
+          fieldId: 'builtin.location',
+          matcher: Matcher.Near,
+          value: {
+            lat: 2,
+            lng: 3,
+            radius: 10
+          },
+        }
+      },
+      {
+        displayName: 'My name',
+        selected: true,
+        filter: {
+          kind: 'fieldValue',
+          fieldId: 'employeeName',
+          matcher: Matcher.Equals,
+          value: 'Bob',
+        }
+      }
+    ]
+  }
+};
+
+const newGeoPosition: GeolocationPosition = {
+  coords: {
+    accuracy: 0,
+    altitude: null,
+    altitudeAccuracy: null,
+    heading: null,
+    latitude: 40.741591687843005,
+    longitude: -74.00530254443494,
+    speed: null,
+  },
+  timestamp: 0
+};
+
+const newGeoPositionWithLowAccuracy: GeolocationPosition = {
+  coords: {
+    ...newGeoPosition.coords,
+    accuracy: 100000,
+  },
+  timestamp: 0
+};
+
+beforeEach(() => {
+  mockAnswersHooks({
+    mockedState,
+    mockedActions: {
+      state: mockedState,
+      setStaticFilters: jest.fn(),
+      executeVerticalQuery: jest.fn()
+    }
+  });
+  jest.spyOn(locationOperations, 'getUserLocation').mockResolvedValue(newGeoPosition);
+});
+
+it('renders custom label when provided', () => {
+  render(<Geolocation label='Click me!' />);
+  const updateLocationButton = screen.getByRole('button', { name: 'Click me!' });
+  expect(updateLocationButton).toBeDefined();
+});
+
+it('renders custom icon when provided', () => {
+  render(<Geolocation GeolocationIcon={() => <img src="graphic1.png" alt="Custom Icon" />} />);
+  const LocationIcon = screen.getByAltText('Custom Icon');
+  expect(LocationIcon).toBeDefined();
+});
+
+it('executes handleClick when provided', async () => {
+  const mockedHandleClickFn = jest.fn();
+  const actions = spyOnActions();
+  render(<Geolocation handleClick={mockedHandleClickFn} />);
+  clickUpdateLocation();
+  await waitFor(() => {
+    expect(mockedHandleClickFn).toHaveBeenCalledWith(newGeoPosition);
+  });
+  expect(actions.executeVerticalQuery).not.toBeCalled();
+});
+
+it('sets a location filter with user\'s coordinates in static filters state when clicked', async () => {
```

Comment:
There are a lot of tests in this file. Could we organize them with some `describe` blocks? For instance, this test and many of the ones below assume there is no `handleClick`. But, the test names don't indicate that. We could provide that additional context with a `describe` block reading `using default click handling` or something.